service/matching/matchingEngine.go

```diff
@@ -167,6 +169,10 @@ func (e *matchingEngineImpl) String() string {
 // if successful creates one.
 func (e *matchingEngineImpl) getTaskListManager(taskList *taskListID,
 	taskListKind *types.TaskListKind) (taskListManager, error) {
+	// Cache user defined task list in taskListCache
+	if *taskListKind == types.TaskListKindNormal {
```

Comment:
can we also add a feature flag here? a little concern about the perf here

Reply:
it might be ok to move this under a read lock

Reply:
Updated with a feature flag.